subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/InitialPassStatementTransformer.java

```diff
@@ -16,39 +16,44 @@
 
 package org.gradle.groovy.scripts.internal;
 
+import org.codehaus.groovy.ast.expr.ArgumentListExpression;
+import org.codehaus.groovy.ast.expr.ConstantExpression;
 import org.codehaus.groovy.ast.stmt.Statement;
 import org.codehaus.groovy.control.SourceUnit;
 import org.codehaus.groovy.syntax.SyntaxException;
 import org.gradle.api.internal.DocumentationRegistry;
 import org.gradle.configuration.ScriptTarget;
-import org.gradle.groovy.scripts.ScriptSource;
-import org.gradle.internal.Factory;
-import org.gradle.plugin.management.internal.PluginRequests;
-import org.gradle.plugin.use.internal.PluginUseScriptBlockMetadataExtractor;
+import org.gradle.plugin.use.internal.PluginUseScriptBlockMetadataCompiler;
 
+import javax.annotation.Nullable;
 import java.util.Arrays;
 import java.util.List;
 
-public class InitialPassStatementTransformer implements StatementTransformer, Factory<PluginRequests> {
+/**
+ * Removes all statements from the given script except the top-level script blocks such as
+ * {@code buildscript}, {@code plugins} and {@code pluginManagement}.
```

Comment:
🎇 🔥 Woot! 🔥 🎇 
Thanks for making sure that this handles `pluginManagement` as well!